be/src/exprs/string_functions.cpp

```diff
@@ -2847,6 +2847,14 @@ static ColumnPtr regexp_replace_use_hyperscan(StringFunctionsState* state, const
         CHECK(false) << "ERROR: Unable to clone scratch space."
                      << " status: " << status;
     }
+    DeferOp op([&] {
+        if (scratch != nullptr) {
+            hs_error_t st;
+            if ((st = hs_free_scratch(scratch)) != HS_SUCCESS) {
+                CHECK(false) << "ERROR: free scratch space failure. status: " << st;
```

Comment:
Please use DCHECK, we can't let BE crash because of one query

Reply:
regexp_replace_use_hyperscan don't support return erorr

Reply:
Fixed